http_proxy.go

```diff
@@ -46,27 +47,18 @@ func main() {
 	}
 
 	// Middleware (runs in reverse order as they are added)
-
-	// Middleware: Forward HTTP Messages
-	forwarder, err := forward.New(nil, forward.IdleTimeoutSetter(time.Duration(*idleClose)*time.Second))
-	if err != nil {
-		log.Error(err)
-	}
-
-	// Middleware: Handle HTTP CONNECT
-	httpConnect, err := httpconnect.New(forwarder, httpconnect.IdleTimeoutSetter(time.Duration(*idleClose)*time.Second))
-	if err != nil {
-		log.Error(err)
-	}
-
-	// Middleware: Common request filter
-	commonHandler, err := commonfilter.New(httpConnect, testingLocal)
-	if err != nil {
-		log.Error(err)
-	}
```

Comment:
Here you can see how easy it is to compose filters into a chain.  I think it's a more fluent API.